src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx

```diff
@@ -1,5 +1,236 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { selectProviderSelectionInfo } from '../../../utils/selectors';
+import ResidentialAddress from '../../../components/ResidentialAddress';
+import { connect } from 'react-redux';
+import * as actions from '../../redux/actions';
+import { FETCH_STATUS } from '../../../utils/constants';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
+import { distanceBetween } from '../../../utils/address';
+import { scrollAndFocus } from '../../../utils/scrollAndFocus';
+import ErrorMessage from '../../../components/ErrorMessage';
 
-export default function ProviderSelectionField() {
-  return <strong>List stuff goes here</strong>;
+const INITIAL_PROVIDER_DISPLAY_COUNT = 5;
+
+function ProviderSelectionField({
+  address,
+  formData,
+  onChange,
+  idSchema,
+  requestStatus,
+  requestProvidersList,
+  communityCareProviderList,
+}) {
+  const [checkedProvider, setCheckedProvider] = useState();
+  const [mounted, setMounted] = useState(false);
+  const [showProvidersList, setShowProvidersList] = useState(false);
+  const [providersListLength, setProvidersListLength] = useState(
+    INITIAL_PROVIDER_DISPLAY_COUNT,
+  );
+  const currentlyShownProvidersList = communityCareProviderList?.slice(
+    0,
+    providersListLength,
+  );
+  const loadingProviders =
+    requestStatus === FETCH_STATUS.loading ||
+    requestStatus === FETCH_STATUS.notStarted;
+  const providerSelected = 'id' in formData;
+  useEffect(() => {
+    requestProvidersList(address);
+    setMounted(true);
+  }, []);
+
+  useEffect(
+    () => {
+      if (showProvidersList && !loadingProviders) {
+        scrollAndFocus('h2');
+      } else if (mounted && !loadingProviders) {
+        scrollAndFocus('.va-button-link');
+      }
+    },
+    [showProvidersList],
```

Comment:
I think you want to add `loadingProviders` in this array, so that if you click on "choose provider" and see the loading state, your focus is set correctly when the loading is finished. Currently, this effect won't fire when loading is done.

Reply:
Fixed